spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/QueueAttributesResolver.java

```diff
@@ -91,8 +94,18 @@ private CompletableFuture<String> resolveQueueUrl() {
 	}
 
 	private CompletableFuture<String> doResolveQueueUrl() {
+		Arn arn = convertToQueueArn(this.queueName);
+		return arn != null
```

Comment:
Well, since we've got this far, we could take a step further and try to handle both calls in the same method, so that we don't duplicate what comes after e.g. `handleException`.

What do you think? Definitely a nit / suggestion, not a blocker.

Reply:
Maybe something along these lines, perhaps better arranged:

```java
	private CompletableFuture<String> doResolveQueueUrl() {
		Arn arn = convertToQueueArn(this.queueName);
		return CompletableFutures.exceptionallyCompose((arn != null
			? this.sqsAsyncClient.getQueueUrl(req -> req.queueName(arn.resourceAsString()).queueOwnerAWSAccountId(arn.accountId().get()))
			: this.sqsAsyncClient.getQueueUrl(req -> req.queueName(this.queueName)))
				.thenApply(GetQueueUrlResponse::queueUrl), this::handleException);
	}
```

Reply:
Thank you @tomazfernandes for the feedback. I also think it makes sense to combine this into one method. I have added a commit with a possible solution.

Reply:
Cool, thanks @techadri!